hassio/src/ingress-view/hassio-ingress-view.ts

```diff
@@ -44,18 +47,101 @@ class HassioIngressView extends LitElement {
 
   private _sessionKeepAlive?: number;
 
+  @state() private _resolveIngressURL: {
+    status?: number;
+    time?: number;
+    interval?: number;
+  } = {};
+
   public disconnectedCallback() {
     super.disconnectedCallback();
 
     if (this._sessionKeepAlive) {
       clearInterval(this._sessionKeepAlive);
       this._sessionKeepAlive = undefined;
     }
+    if (this._resolveIngressURL.interval) {
+      clearInterval(this._resolveIngressURL.interval);
+    }
   }
 
-  protected render(): TemplateResult {
+  public connectedCallback() {
+    super.connectedCallback();
+
+    this._resolveURL();
+    this._resolveIngressURL.interval = window.setInterval(async () => {
+      await this._resolveURL();
+    }, 1000);
+  }
+
+  private async _resolveURL(): Promise<void> {
     if (!this._addon) {
-      return html` <hass-loading-screen></hass-loading-screen> `;
+      return;
+    }
+
+    if (this._addon.state !== "started") {
+      clearInterval(this._resolveIngressURL.interval);
+      await showAlertDialog(this, {
+        text:
+          this.hass.localize("supervisor_ingress.not_running") ||
+          "The add-on is not running, please start it.",
+        title: this._addon.name,
+        confirmText:
+          this.hass.localize("supervisor_ingress.go_to_dashboard") ||
+          "Go to add-on dashboard",
+      });
+      await nextRender();
+      navigate(`/hassio/addon/${this._addon.slug}/info`, { replace: true });
+      return;
+    }
+
+    if (this._resolveIngressURL.status !== STATUS_BAD_GATEWAY) {
+      if (this._resolveIngressURL.interval) {
+        clearInterval(this._resolveIngressURL.interval);
+      }
+    }
+
+    if (
+      this._resolveIngressURL.time &&
+      new Date().getTime() > this._resolveIngressURL.time + TIMEOUT
+    ) {
+      await showAlertDialog(this, {
+        text:
+          this.hass.localize("supervisor_ingress.timeout") ||
+          "Timeout while waiting for add-on to start, check the add-on logs",
+        title: this._addon.name,
+        confirmText:
+          this.hass.localize("supervisor_ingress.go_to_logs") ||
+          "Go to add-on logs",
+      });
+      await nextRender();
+      navigate(`/hassio/addon/${this._addon.slug}/logs`, { replace: true });
+      return;
+    }
+
+    try {
+      const response = await fetch(this._addon.ingress_url!);
+      this._resolveIngressURL.status = response.status;
```

Comment:
Either make a copy of `_resolveIngressURL` or break them up into separate properties.

Reply:
But it is actually only `status` that we want as a `state`

Reply:
no need